### PR TITLE
Add 2.0.0-rc1 release notes

### PR DESCRIPTION
**Pull request by Poojita-Raj**
Signed-off-by: Poojita Raj <poojiraj@amazon.com>

### Description
Adding release notes for the 2.0 release. There was no common branch point to start the release notes from - so it started from commit f14e0a36c48ecb8d2f58b37cc18e86d3b6836319 - the commit after #2489. Dependabot PRs have been removed for the sake of brevity.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by nknize — CHANGES_REQUESTED** [review]
Rebase from main to ensure passing tests.

**Review by dblock — CHANGES_REQUESTED** [review]
These notes need to be humanly readable. This isn't really. So I am afraid we need to hand-edit this entire file.

Screen Shot 2022-04-22 at 5 34 21 PM

@elfisher @bbarani can agree/disagree

**Comment by dblock** [comment]
Check out what https://github.com/opensearch-project/job-scheduler/pull/172 looks like, I think I expected something like that here.

File: release-notes/opensearch.release-notes-2.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,4784 @@
+## Version 2.0.0-rc1 Release Notes
+
+* __Add type mapping removal bwc tests for indexing, searching, snapshots (#2901)__
```
Comment (dblock) [inline release-notes/opensearch.release-notes-2.0.0-rc1.md:3]: I'd want those issues to be clickable.

**Review by dblock — CHANGES_REQUESTED** [review]
I would just keep the titles and make the GitHub issues clickable. Like in https://github.com/opensearch-project/job-scheduler/pull/172/files.

**Review by dblock — COMMENTED** [review]
Much better! Do you think it's worth doing a pass on aligning language/standardizing things like `[Remove] Xyz ...`, capitalizing consistently, etc.?

**Comment by Poojita-Raj** [comment]
> Much better! Do you think it's worth doing a pass on aligning language/standardizing things like `[Remove] Xyz ...`, capitalizing consistently, etc.?

I don't think it's worth it since a lot of the titles would require a manual rewrite to ensure they're in the same format - not all of them start with action verbs or tags. But going forward definitely we should ensure all PRs have a check to ensure they're in the right format - referencing the discussion in #1868.

**Review by kartg — APPROVED** [review]
Kudos for combing through all of the commits and crafting this!

**Pull request merged**